### PR TITLE
Hs 43 comments

### PR DESCRIPTION
**Pull request by nadavsofi**

### CODE REVIEW

File: src/components/Comments.vue [hunk 1]
```diff
@@ -222,7 +380,19 @@ export default class Comments extends Vue {
   }
 
   get rootComments() {
-    return this.comments.filter(comment => comment.parent == null);
+    return this.comments.filter(
+      comment => comment.parent == null && comment.isHidden != true
```
Comment (avirankatzsofi) [inline src/components/Comments.vue]: `!= true` is redundant. Should be `!comment.isHidden`

File: src/components/Comments.vue [hunk 2]
```diff
@@ -222,7 +380,19 @@ export default class Comments extends Vue {
...
+    );
+  }
+
+  childComments(parentComment) {
+    return parentComment.children.filter(
+      childComment => childComment.isHidden != true
```
Comment (avirankatzsofi) [inline src/components/Comments.vue]: `!= true` is redundant. Should be `!comment.isHidden`

File: src/helpers/mutations.js [hunk 3]
```diff
@@ -182,10 +182,12 @@ export const createComment = `mutation createComment(
   $plan: ID!
   $parent: ID
   $user: ID
+  $isPinned: Boolean
```
Comment (avirankatzsofi) [inline src/helpers/mutations.js]: redundant variable (always `false`)

File: src/helpers/mutations.js [hunk 4]
```diff
@@ -182,10 +182,12 @@ export const createComment = `mutation createComment(
...
+  $isHidden: Boolean
 ) {
   createComment(
     input: {
-      data: { title: $title, name: $name, content: $content, plan: $plan, user: $user, parent: $parent }
+      data: { title: $title, name: $name, content: $content, plan: $plan, user: $user, parent: $parent, isPinned: $isPinned, isHidden: $isHidden }
```
Comment (avirankatzsofi) [inline src/helpers/mutations.js]: replace `isPinned: $isPinned, isHidden: $isHidden` with `isPinned: false, isHidden: false`

File: src/helpers/mutations.js [hunk 5]
```diff
@@ -221,11 +246,47 @@ export const updateMyPlan = `mutation updateMyPlan(
         sections: $sections
         attachedFiles: $files
         lastUpdate: $update
+        lockComments: $lockComments
       }
     }
   ) {
     plan {
       id
+      sid
```
Comment (avirankatzsofi) [inline src/helpers/mutations.js]: double check which fields are necessary

File: src/views/Plan.vue [hunk 6]
```diff
@@ -123,10 +156,43 @@ import SubscriptionToggle from "../components/SubscriptionToggle.vue";
...
   @Getter(Getters.SELECTED_PLAN) plan;
   /** @type {import("../../graphql/types").Meeting[]} */
   @Getter(Getters.MANAGABLE_MEETINGS) managableMeetings;
+  @Action(ActionTypes.UPDATE_PLAN) updatePlanAction;
+
+  loader = false;
+
+  comments = {
+    locked: "",
+    switch: "",
```
Comment (avirankatzsofi) [inline src/views/Plan.vue]: Omit switch, use locked only

**Review by avirankatzsofi — APPROVED** [review]

**Review by avirankatzsofi — CHANGES_REQUESTED** [review]
As specified in the comments

File: src/components/Comments.vue (hunk 1 above)
Comment (nadavsofi) [reply src/components/Comments.vue]: fixed

File: src/components/Comments.vue (hunk 2 above)
Comment (nadavsofi) [reply src/components/Comments.vue]: fixed

File: src/helpers/mutations.js (hunk 3 above)
Comment (nadavsofi) [reply src/helpers/mutations.js]: fixed

File: src/helpers/mutations.js (hunk 4 above)
Comment (nadavsofi) [reply src/helpers/mutations.js]: fixed

File: src/helpers/mutations.js (hunk 5 above)
Comment (nadavsofi) [reply src/helpers/mutations.js]: fixed

File: src/views/Plan.vue (hunk 6 above)
Comment (nadavsofi) [reply src/views/Plan.vue]: fixed

File: src/views/Plan.vue [hunk 7]
```diff
@@ -99,25 +99,40 @@
...
             >
-            </v-switch>
+              <v-icon small left>mdi-lock</v-icon>
+              ההתייחסויות נעולות
+            </v-btn>
           </v-col>
           <v-col
-            v-else-if="comments.locked && !currentUserIsCommentsAdmin"
+            v-else-if="
+              planData.commentsAreLocked && !currentUserIsCommentsAdmin
```
Comment (avirankatzsofi) [inline src/views/Plan.vue]: should work without `&& !currentUserIsCommentsAdmin` because we're inside else-if

File: src/views/Plan.vue [hunk 8]
```diff
@@ -196,16 +200,26 @@ export default class Plan extends Vue {
...
     const res = await this.updatePlanAction(this.planData);
-    this.lockCommentLoader = !res;
+    if (!res.status) {
+      this.lockCommentErrMessage = res.message;
+    } else {
+      this.planData.commentsAreLocked = false;
+      this.lockCommentErrMessage = "";
+    }
+    this.lockCommentLoader = false;
   }
```
Comment (avirankatzsofi) [inline src/views/Plan.vue]: Duplicate code - can be refactored into a single function, such as `switchCommentsLockState`

**Review by avirankatzsofi — CHANGES_REQUESTED** [review]

**Review by avirankatzsofi — APPROVED** [review]

**Review by avirankatzsofi — APPROVED** [review]

**Pull request merged**